woocommerce/woocommerce-android
# Display campaign created success bottom sheet

Closes: #10736


### Description

Displays a campaign-created success bottom sheet after successfully creating a Blaze campaign from the native flow: 

| Light  | Dark |
| ------------- | ------------- |
| <img width="300"  src="https://github.com/woocommerce/woocommerce-android/assets/2663464/09038ce6-4aa1-4f1f-9958-ddf85d0e2a2c">   | <img width="300"  src="https://github.com/woocommerce/woocommerce-android/assets/2663464/91aeedc1-4a97-427c-8c16-0ed9bd5728c1">   |

A few notes about the implementation. 
The Blaze campaign creation flow can be triggered from different places across the app: my store, product detail, campaigns list, the more menu. To avoid having to pass an `origin` value through all the Blaze screens so we know what screen of the app we should navigate to upon successfully creating the campaign, I `popup` all the Blaze campaign screens from the stack and the user will land on the screen that originated the flow. 
With this approach, to be able to show the bottom sheet correctly, I had to add a new shared pref property `isComingFromBlazeCampaignCreationSuccess` that is then read from any of the triggering Blaze flow fragments to launch the bottom sheet accordingly. I don't know if there's a better way to do this. Ideally, we'd pass an argument back to the origin fragment, but I haven't found a way to `popUp` several fragments from the stack and at the same time pass an argument. 

### Testing instructions

1. Trigger Blaze campaign creation flow from My Store tab
2. Click on the `Promote` button
3. Click on `Submit Campaign`
4. Notice how you navigate back to My Store tab and the success bottom sheet is displayed

Repeat the steps starting the flow from product detail and from campaign list

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/2663464/4dfb6112-5005-47eb-86e2-43dc14222b1c
